src/macaron/repo_finder/repo_finder_dd.py

```diff
@@ -0,0 +1,203 @@
+        version : str
+            The version number of an artifact.
+
+        Yields
+        ------
+        Iterator[str] :
+            The URLs found for the passed GAV.
+        """
+        request_urls = self.create_urls(group, artifact, version)
+        if not request_urls:
+            logger.debug("No urls found for: %s", artifact)
+            return
+
+        with requests.Session() as session:
+            metadata = self.retrieve_metadata(session, request_urls[0])
+            if not metadata:
+                logger.debug("Failed to retrieve metadata for: %s", artifact)
+                return
+
+        urls = self.read_metadata(metadata)
+        if not urls:
+            logger.debug("Failed to extract repository URLs from metadata: %s", artifact)
+            return
+
+        yield from iter(urls)
+
+    def create_urls(self, group: str, artifact: str, version: str) -> list[str]:
+        """
+        Create the urls to search for the metadata relating to the passed artifact.
+
+        If a version is not specified, remote API calls will be used to try and find one.
+
+        Parameters
+        ----------
+        group : str
+            The group ID.
+        artifact: str
+            The artifact ID.
+        version: str
+            The version of the artifact.
+
+        Returns
+        -------
+        list[str]
+            The list of created URLs.
+        """
+        package_name = f"{group}:{artifact}@{version}"
+        base_url = self.create_type_specific_url(group, artifact)
+
+        if version:
+            return [f"{base_url}/versions/{version}"]
+
+        # Find the latest version.
+        with requests.Session() as session:
+            try:
+                result = session.get(base_url)
+            except (requests.RequestException, OSError) as error:
+                logger.debug("Error during version retrieval: %s", error)
+                return []
+
+            if not result.ok:
+                logger.debug("Failed to retrieve versions for: %s, error code: %s", package_name, result.status_code)
+                return []
+
+        metadata = json.loads(result.text)
+        versions = metadata["versions"]
+        latest_version = versions[len(version) - 1]["versionKey"]["version"]
+
+        if latest_version:
+            return [f"{base_url}/versions/{latest_version}"]
+
+        return []
+
+    def retrieve_metadata(self, session: requests.Session, url: str) -> str:
+        """
+        Attempt to retrieve the file located at the passed URL using the passed Session.
+
+        Parameters
+        ----------
+        session : requests.Session
+            The HTTP session to use for attempting the GET request.
+        url : str
+            The URL for the GET request.
+
+        Returns
+        -------
+        str :
+            The retrieved file data or an empty string.
+        """
+        try:
+            result = session.get(url)
```

Comment:
To improve handling errors for requests, let's use the [send_get_http](https://github.com/oracle/macaron/blob/fa2622b588546888041cbf3a0c07a67841a98080/src/macaron/util.py#L21) function. As discussed, we will revisit retrying requests when there is a timeout [separately](https://github.com/oracle/macaron/issues/398).

Reply:
I've changed this and the other similar calls of the Repo Finder. I chose to use send_get_http_raw as it fits with the class design better, and the functionality seems to be the same.